AWSAuthSDK/Sources/AWSMobileClient/AWSMobileClient.swift

```diff
@@ -88,6 +88,8 @@ final public class AWSMobileClient: _AWSMobileClient {
     var userPassword: String? = nil
 
     var tokenOperations:NSHashTable<FetchUserPoolTokensOperation> = NSHashTable.weakObjects()
+    let tokenOperationsQueue: DispatchQueue = DispatchQueue(label:"AWSMobileClient.tokenOperationsQueue",
```

Comment:
There is similarly named queue `tokenFetchOperationQueue` and can be confusing. Should we rename this `queueToAccessTokenOperations` or some other name?

Reply:
Agreed, it becomes confusing.
So I instead decided to encapsulate this behaviour inside a new `WeakHashTable` class